tests/functional/context/types/test_type_from_abi.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+from vyper.context.types import get_pure_types
+from vyper.context.types.indexable.sequence import ArrayDefinition
+from vyper.context.types.utils import get_type_from_abi
+from vyper.exceptions import UnknownType
+
+BASE_TYPES = ["int128", "uint256", "bool", "address", "bytes32"]
+
+
+@pytest.mark.parametrize("type_str", BASE_TYPES)
+def test_base_types(namespace, type_str):
+    pure_type = get_pure_types()[type_str]
```

Comment:
Man, I really don't like the term "pure types", it's very strange.

Basic? Base? Built-in?

Reply:
This will touch a lot of files.  Once this PR is good to merge, I'll make the name change in a new one.

Reply:
Sounds good!